astropy/tests/runner.py

```diff
@@ -251,7 +251,12 @@ def run_tests(self, **kwargs):
         # also harmless to nest the contexts
         with set_temp_config(astropy_config, delete=True):
             with set_temp_cache(astropy_cache, delete=True):
-                return pytest.main(args=args, plugins=plugins)
+                retv = pytest.main(args=args, plugins=plugins)
+
+        import hypothesis
+
+        hypothesis.settings.load_profile("astropy_repeat")
+        return retv
```

Comment:
This looks pretty sensible to me!  My best guess, given that it's still failing, is that the load-profile in your conftest.py is exectued again and thus overrides this.

One possible workaround would be to monkeypatch some attribute onto `astropy.testing` here, and then make the load in `conftest.py` conditional on that _not_ being set.

Reply:
I don't think messing with conftest would help, as we rely on it to also reset some other things essential to the test suite. If injecting CLI isn't an option and this does not work easily, then I'd say we try the env var route as we had done with `IS_CRON` and so on. We can have, say, `IS_DOUBLE_RUN` set just for the `double` job. The trick is passing it in correctly via tox into OpenAstronomy reusable workflow. So best to also print out the value during test run just so we can check. I think grepping `IS_CRON` would give you a pretty good idea how we managed to get that one to work.

Thanks!

Reply:
🤦🏻‍♂️ I didn't notice we already had a call to `load_profile` in `conftest.py`, thanks a lot for pointing it out !
Monkeypatching as you're suggesting would work, but would also create more complexity than I think we want to be handling here. I'm going to try and add these settings to our default profile *unconditionally* and see how others like it.

Reply:
@pllim for completion, I think we were again typing at the same time so I didn't see your reply. I feel like it's probably best to keep it as simple as possible so in my opinion #17301 or unconditional and global filtering of this healthcheck are better alternatives, but let me know if you prefer I make it more subtle. Thanks !